src/WebJobs.Script/Host/HostIdValidator.cs

```
@@ -133,7 +134,11 @@ internal async Task WriteHostIdAsync(string hostId, HostIdInfo hostIdInfo)
         {
             try
             {
-                var containerClient = _storageProvider.GetBlobContainerClient();
+                if (!_azureBlobStorageProvider.TryCreateHostingBlobContainerClient(out var containerClient))
+                {
+                    throw new InvalidOperationException($"Could not create Hosting BlobContainerClient in {nameof(WriteHostIdAsync)}");
```

Just a note that these methods (WriteHostIdAsync and ReadHostIdInfoAsync are not called directly except for tests, so this throws should not be triggered. ValidateHostIdUsageAsync will silently fail if _azureBlobStorageProvider.TryCreateHostingBlobContainerClient() fails.

---

They are called directly a few lines above. What you should do is capture the client created on line 71 and pass that into these method calls, so we don't have to duplicate the calls. It also solves your error problem here.